File: 05/Rigon/01.c
/* 1. Desenvolva um programa em C que leia um conjunto X com N elementos reais e
que calcule a diferença entre o maior e o menor elemento existente, além de
indicar as posições em que eles ocorrem. */

#include <stdio.h>
#include <stdlib.h>

int main(){
  int N, i, posma, posme;
  float dif = 0.0, maior, menor;
  printf("Digite a quantidade de elementos a serem calculados\n");
  scanf("%d", &N);
  float X[N];

  for(i=0;i < N;i++){
    scanf("%f", &X[i]);

    if ( i == 0){
      maior = X[i];
      menor = X[i];
    }

    if ( X[i] > maior){
      maior = X[i];
      posma = i;
    }

    if ( X[i] < menor){
      menor = X[i];
      posme = i;
    }
  }

  dif = maior - menor;
  printf("Maior %.2f \tposicao %d\n", maior, posma);
  printf("Menor %.2f \tposicao %d\n", menor, posme);
  printf("Diferenca: %.2f \n", dif);

  return 0;
}

File: 05/Rigon/python/01.py
# -*- coding: utf-8 -*-
'''
1. Desenvolva um programa em C que leia um conjunto X com N elementos reais e
que calcule a diferença entre o maior e o menor elemento existente, além de
indicar as posições em que eles ocorrem.
'''

N = int(input("Digite um valor:"))
print(N)
X = []

for i in range(N):
    num = float(input("Digite uma nota:"))
    X.append(num)

for i in range(N):
    if (i == 0):
        maior = X[i]
        menor = X[i]

    if(X[i] >= maior):
        maior = X[i]
        posmaior = i

    if(X[i] <= menor):
        menor = X[i]
        posmenor = i

dif = maior - menor
print("Maior valor: {:.2f}, na posição {:d}" .format(maior, posmaior))
print("Menor valor: {:.2f}, na posição {:d}" .format(menor, posmenor))
print("Diferença dos valores: '{:.2f}'" .format(dif))
